src/python/m5/params.py

```diff
@@ -587,19 +595,22 @@ def __init__(cls, name, bases, dict):
         if name == "CheckedInt":
             return
 
-        if not (hasattr(cls, "min") and hasattr(cls, "max")):
-            if not (hasattr(cls, "size") and hasattr(cls, "unsigned")):
+        if not hasattr(self, "min") or not hasattr(self, "max"):
+            if (size := getattr(self, "size", None)) is not None:
+                # TODO: Maybe only override min and max if they are not already set
+                # (rather than overriding both if either are unset)
+                if getattr(self, "unsigned", False):
+                    self.min = 0
+                    self.max = 2**size - 1
+                else:
+                    self.min = -(2 ** (size - 1))
+                    self.max = (2 ** (size - 1)) - 1
+            else:
```

Comment:
Adding a comment so we can discuss this TODO, does that sound like a good idea?
It seems right, but it changes behavior, so I held off doing it